lib/logstash/inputs/kafka.rb

```diff
@@ -335,6 +343,9 @@ def do_poll(consumer)
     rescue org.apache.kafka.common.errors.WakeupException => e
       logger.debug("Wake up from poll", :kafka_error_message => e)
       raise e unless stop?
+    rescue org.apache.kafka.common.errors.FencedInstanceIdException => e
+      logger.error("Another consumer with same group.instance.id has connected")
```

Comment:
I think we could just add the `:group_instance_id` but in case of `consumer_threads` is > 1 that would be inaccurate and maybe it's better to add the exception message. Other, we should propagate the effective  `group_instance_id` down to the `#do_poll` method.
WDYT?

Reply:
you can abstract away the value with a method, something like this:

```ruby
def group_instance_id_w_threads
  return unless group_instance_id
  consumer_threads > 1 ? "#{group_instance_id}-#{i}" : group_instance_id
end
```

Reply:
Sorry, I don't understand how it solves the problem of propagate the "number of thread" down to the `#do_poll`.

The message of the caught `org.apache.kafka.common.errors.FencedInstanceIdException` exception is something like hte following:
```
Get fenced exception for group.instance.id test_group_id, current member.id is test_group_id-539e9f45-f91f-4317-93d5-20cd78dddbc6
```

so it already contains the effective `group.instance.id ` used by the consumer instance.
We could print it, but in that case we bind to the exception, and if in a future version of the Kafka library that information is removed we also lost it. I think it's unlikely that it happens, but we would rely on the client exception to provide information that's valuable for the customer.

Reply:
`do_poll` has access to `consumer_threads` since it's a setting. but if the exception message already contains the id then nevermind :)

Reply:
Using the exception message the error log becomes:
```
"Another consumer with same group.instance.id has connected", {:original_error_message=>"The broker rejected this static consumer since another consumer with the same group.instance.id has registered with a different member.id."}
```
which helps  more than before in trouble shooting possible problems.